File: demo/java/src/main/java/info/chenliang/tank/Client2ServerDispatcher.java
package info.chenliang.tank;
import java.io.*;
import info.chenliang.talky.*;
public class Client2ServerDispatcher {
public static void dispatch(DataInputStream dis, Client2ServerProxy proxy) throws Exception{
int fid = (int)(dis.read()&0xFFFF);
switch(fid)
{
case 0:
{
fire(dis, proxy);
}
break;
case 1:
{
useItem(dis, proxy);
}
break;
}
}
protected static void fire(DataInputStream dis, Client2ServerProxy proxy) throws Exception{
byte playerId;
playerId = dis.readByte();
proxy.fire(playerId);
}
protected static void useItem(DataInputStream dis, Client2ServerProxy proxy) throws Exception{
byte seq;
seq = dis.readByte();
proxy.useItem(seq);
}
}

File: demo/java/src/main/java/info/chenliang/tank/Client2ServerDispatcher.cs
namespace info.chenliang.tank{
using System;
using System.IO;
using MiscUtil;
using MiscUtil.IO;
using MiscUtil.Conversion;
using Talky;
public class Client2ServerDispatcher {
public static void Dispatch(MemoryEndianBinaryReader reader, Client2ServerProxy proxy) {
int fid = reader.ReadByte();
switch(fid)
{
case 0:
{
fire(reader, proxy);
}
break;
case 1:
{
useItem(reader, proxy);
}
break;
}
}
protected static void fire(MemoryEndianBinaryReader reader, Client2ServerProxy proxy) throws Exception{
sbyte playerId;
playerId = reader.ReadByte();
proxy.fire(playerId);
}
protected static void useItem(MemoryEndianBinaryReader reader, Client2ServerProxy proxy) throws Exception{
sbyte seq;
seq = reader.ReadByte();
proxy.useItem(seq);
}
}
}
